nav2_map_server/include/nav2_map_server/map_3d/map_server_3d.hpp

```diff
@@ -0,0 +1,152 @@
+   * @return Success or Failure
+   */
+  nav2_util::CallbackReturn on_deactivate(const rclcpp_lifecycle::State & state) override;
+  /**
+   * @brief Resets the member variables
+   * @param state Lifecycle Node's state
+   * @return Success or Failure
+   */
+  nav2_util::CallbackReturn on_cleanup(const rclcpp_lifecycle::State & state) override;
+  /**
+   * @brief Called when in Shutdown state
+   * @param state Lifecycle Node's state
+   * @return Success or Failure
+   */
+  nav2_util::CallbackReturn on_shutdown(const rclcpp_lifecycle::State & state) override;
+
+  /**
+   * @brief Load the map YAML, image from map file name and
+   * generate output response containing a PointCloud2.
+   * Update pcd_msg_ class variable.
+   * @param yaml_file name of input YAML file
+   * @param response Output response with loaded PointCloud2 map
+   * @return true or false
+   */
+  bool loadMapResponseFromYaml(
+    const std::string & yaml_file,
+    std::shared_ptr<nav2_msgs::srv::LoadMap3D::Response> response);
+
+  /**
+   * @brief Map getting service callback for pcd
+   * @param request_header Service request header
+   * @param request Service request
+   * @param response Service response
+   */
+  void getMapCallback(
+    const std::shared_ptr<rmw_request_id_t> request_header,
+    const std::shared_ptr<nav2_msgs::srv::GetMap3D::Request> request,
+    std::shared_ptr<nav2_msgs::srv::GetMap3D::Response> response);
+
+  /**
+   * @brief Map loading service callback for pcd
+   * @param request_header Service request header
+   * @param request Service request
+   * @param response Service response
+   */
+  void loadMapCallback(
+    const std::shared_ptr<rmw_request_id_t> request_header,
+    const std::shared_ptr<nav2_msgs::srv::LoadMap3D::Request> request,
+    std::shared_ptr<nav2_msgs::srv::LoadMap3D::Response> response);
+
+  void updateMsgHeader();
+
+  // The name of the service for getting a map
+  const std::string service_name_{"map"};
+
+  // The name of the service for loading a map
+  const std::string load_map_service_name_{"load_map"};
+
+  // PointCloud2 related fields
+
+  // A service to provide the pointcloud2 (GetMap3D) and the message to return
+  rclcpp::Service<nav2_msgs::srv::GetMap3D>::SharedPtr pcd_service_;
+
+  // A service to load the PointCloud2 from file at run time (LoadMap3D)
+  rclcpp::Service<nav2_msgs::srv::LoadMap3D>::SharedPtr pcd_load_map_service_;
+
+  // A topic on which the PointCloud2 will be published
+  rclcpp_lifecycle::LifecyclePublisher<sensor_msgs::msg::PointCloud2>::SharedPtr pcd_pub_;
+
+  // The frame ID used in the returned PointCloud2 message
+  std::string frame_id_;
+
+  // The message to publish the pointcloud topic
+  sensor_msgs::msg::PointCloud2 pcd_msg_;
+
+  // A topic on which the Pose will be published
+  rclcpp_lifecycle::LifecyclePublisher<geometry_msgs::msg::Pose>::SharedPtr origin_pub_;
+
+  // The message to publish the pose topic
+  geometry_msgs::msg::Pose origin_msg_;
```

Comment:
Arrange PCD and origin publishers/message variables. Currently they are scattered in the code.